src/njoy21/input/HEATR/test/HEATR.test.cpp

```diff
@@ -0,0 +1,213 @@
+        "1325 0 1 0 0 2 27\n"
+        "16\n"
+	"55e6\n"
+        ) );
+      HEATR heatr( iss );
+      THEN( "all cards have the correct values" ){
+        REQUIRE( heatr.card1.nendf.value == -21 );
+        REQUIRE( heatr.card1.nin.value   == -22 );
+        REQUIRE( heatr.card1.nout.value  == -24 );
+        REQUIRE( heatr.card1.nplot.value == -25 );
+
+	REQUIRE( heatr.controlTuple );
+
+        const auto& card4 = std::get<0>( heatr.controlTuple.value() );
+	REQUIRE( card4.mta.value[0] == 16 );
+
+        const auto& card5 = std::get<1>( heatr.controlTuple.value() );
+	REQUIRE( card5.qa.value[0] == 55e6 * dimwits::electronVolts);
+      } // THEN
+    } // WHEN
+
+    WHEN( "Cards 3, 4, and 5 are invoked, qa values < 99e6 eV" ){
+      iRecordStream<char> iss( std::istringstream( 
+        "-21 -22 -24 -25\n"
+        "1325 2 2 0 0 2 27\n"
+	"444 445\n"
+        "16 18\n"
+	"55e6 56e6\n"
+        ) );
+      HEATR heatr( iss );
+      THEN( "all cards have the correct values, no card5a values" ){
+        REQUIRE( heatr.card3->mtk.value.size() == 2 );
+	REQUIRE( heatr.card3->mtk.value[0] == 444 );
+	REQUIRE( heatr.card3->mtk.value[1] == 445 );
+	
+	const auto& card4 = std::get<0>( heatr.controlTuple.value() );
+	REQUIRE( card4.mta.value.size() == 2 );
+	REQUIRE( card4.mta.value[0] == 16 );
+	REQUIRE( card4.mta.value[1] == 18 );
+
+	const auto& card5 = std::get<1>( heatr.controlTuple.value() );
+	REQUIRE( card5.qa.value.size() == 2 );
+	REQUIRE( card5.qa.value[0] == 55e6 * dimwits::electronVolts );
+	REQUIRE( card5.qa.value[1] == 56e6 * dimwits::electronVolts );
+
+        REQUIRE( std::get<2>( heatr.controlTuple.value() ).size() == 0 );
+      } // THEN
+    } // WHEN
+
+    WHEN( "Cards 4 and 5 are invoked, qa values > 99e6 eV" ){
+      iRecordStream<char> iss( std::istringstream( 
+        "-21 -22 -24 -25\n"
+        "1325 0 2 0 0 2 27\n"
+        "16 18\n"
+	"55e6 99e6\n"
+	"0 0 0 0 1 3\n"
+	"3 2\n"
+	"8.9e6 -8.14e6 9.0e6 -8.14e6 1.1e7 -8.36e6\n"
+        ) );
+      HEATR heatr( iss );
+      THEN( "all cards have the correct values, along with card5a values" ){
+	const auto& card4 = std::get<0>( heatr.controlTuple.value() );
+	REQUIRE( card4.mta.value.size() == 2 );
+	REQUIRE( card4.mta.value[0] == 16 );
+	REQUIRE( card4.mta.value[1] == 18 );
+
+	const auto& card5 = std::get<1>( heatr.controlTuple.value() );
+	REQUIRE( card5.qa.value.size() == 2 );
+	REQUIRE( card5.qa.value[0] == 55e6 * dimwits::electronVolts );
+	REQUIRE( card5.qa.value[1] == 99e6 * dimwits::electronVolts );
+	
+	const auto& card5aList = std::get<2>( heatr.controlTuple.value() );
+	REQUIRE( card5aList.size() == 1 );
+	REQUIRE( card5aList[0].qbar.value.NR() == 1 );
+	REQUIRE( card5aList[0].qbar.value.NP() == 3 );
+	std::vector<double> x{8.9e6, 9.0e6, 1.1e7};
+	std::vector<double> y{-8.14e6, -8.14e6, -8.36e6};
+	REQUIRE( ranges::equal( x, card5aList[0].qbar.value.x() ) );
+	REQUIRE( ranges::equal( y, card5aList[0].qbar.value.y() ) );
+      } // THEN
+    } // WHEN
+
```

Comment:
Please add a test with multiple card5a's (unless it's already there and I overlooked it)